docs/usage.md

```diff
@@ -0,0 +1,99 @@
+```
+cd /home/username/cc/database
+postgres -D . -p 6250
+```
+
+### Create database and tables
+```
+createdb tinyxml -p 6250
+cd /home/username/cc/CodeCompass
+find -name *.sql | while read -r line; do psql -f $line -d tinyxml -p 6250 -h localhost; done
+```
+
+### Prepare an example project for parse
+```
+# Create a project folder
+mkdir -p /home/username/cc/projects
+cd /home/username/cc/projects
+
+# Clone an example cmake based project
+git clone https://github.com/dartsim/tinyxml
+mkdir -p tinyxml/build
+cd tinyxml/build
+```
+
+### Generate compilation databse
+If you want to parse a C++ project, you have to create a compilation database file. You can create this by cmake with the option CMAKE_EXPORT_COMPILE_COMMANDS (http://clang.llvm.org/docs/JSONCompilationDatabase.html). 
+```
+# Generate compilation database with cmake
+cmake .. -DCMAKE_EXPORT_COMPILE_COMMANDS=1
+```
+*Note:* it will be automatized later.
+
+## 1. Parse a project
+For parsing a project with CodeCompass, the following command has to be emitted:
+
+```
+CodeCompass_parser --data-dir <datadir> -n <name> -i <input1> -i <input2> -d <connection_string>
+```
+- **Data directory**: This is a directory where the parsed project and different config and log files are located.
+- **Input**: Several inputs can be given with -i flags. An input can be a directory or a compilation database. The plugins iterate these inputs and decide if it can use it. For example the C++ parser will consume the given compilation databases and the text search parser will consume the files under the given directories. It is entirely up to the parser what it does with the input parameters.
+- **Database**: The plugins can use an SQL database as storage. By the connection string the user can give the location of a running database system. 
+
+### Copy clang include directory to install
+```
+cp -avr /home/username/cc/CodeCompass-deps/lib/clang /home/username/cc/CodeCompass-install/lib/
+```
+
+### Create workspace config file in the workspace directory: 
+```
+vim /home/username/cc/workdir/workspace.cfg
+```
+The content of this file:
+```
+[workspace.tinyxml]
+connection = pgsql:host=localhost;database=tinyxml;user=user_name;port=6250
+description = Tinyxml project
+datadir = /home/username/cc/workdir/CodeCompass/data
+```
+*Note:* it will be automatized later.
+
+### Using example:
+```
+# Parse the prepared tinyxml project
+./home/username/cc/CodeCompass-install/bin/CodeCompass_parser -d "pgsql:database=tinyxml;port=6250" --data-dir /home/username/cc/workdir/tinyxml/data -i /home/username/cc/projects/tinyxml/build/compile_commands.json -j1
+```
+
+## 2. Start the webserver
+You can start the CodeCompass weserver with bin/CodeCompass_webserver binary in the CodeCompass installation directory.
+
+```
+CodeCompass_webserver -w <workdir> -p <port> -d <connection_string>
+```
+- **Workspace**: This is a directory where the parsed projects and different config and log files are located. 
```

Comment:
For the first sight it might not be too intuitive that the `workspace.cfg` file has to be located in the `workspace` directory.